src/main/java/de/dennisguse/opentracks/services/TrackRecordingService.java

```diff
@@ -384,20 +384,31 @@ private void showNotification(boolean isGpsStarted) {
         if (isRecording()) {
             Intent intent = IntentUtils.newIntent(this, TrackRecordingActivity.class)
                     .putExtra(TrackRecordingActivity.EXTRA_TRACK_ID, recordingStatus.getTrackId());
+
+            int pendingIntentFlags = PendingIntent.FLAG_UPDATE_CURRENT;
+            if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.M) {
+                pendingIntentFlags |= PendingIntent.FLAG_IMMUTABLE;
+            }
             PendingIntent pendingIntent = TaskStackBuilder.create(this)
                     .addParentStack(TrackRecordingActivity.class)
                     .addNextIntent(intent)
-                    .getPendingIntent(0, PendingIntent.FLAG_UPDATE_CURRENT);
+                    .getPendingIntent(0, pendingIntentFlags);
 
             notificationManager.updatePendingIntent(pendingIntent);
             notificationManager.updateContent(getString(R.string.gps_starting));
             startForeground(TrackRecordingServiceNotificationManager.NOTIFICATION_ID, notificationManager.getNotification());
         }
         if (!isRecording() && isGpsStarted) {
             Intent intent = IntentUtils.newIntent(this, TrackListActivity.class);
-            PendingIntent pendingIntent = TaskStackBuilder.create(this)
+            PendingIntent pendingIntent = null;
+
+            int pendingIntentFlags = 0;
+            if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.M) {
+                pendingIntentFlags = PendingIntent.FLAG_IMMUTABLE;
+            }
+            pendingIntent = TaskStackBuilder.create(this)
```

Comment:
PendingIntent can be declared here as well.

Reply:
Done :)